cloud/terraform/terraform_controller.py

```diff
@@ -0,0 +1,105 @@
+from ast import alias
+import os
+import json
+
+
+class TerraformController:
+    def __init__(self, cloud):
+        self.cloud = cloud
+
+        if cloud == 'aws':
+            self.cloud_instance = 'aws_instance'
+        self.resources = {'resource': {self.cloud_instance: {}}}
+        self.providers = {'provider': {cloud: []}}
+
+    def __del__(self):
+        self.destroy_infra()
+
+    def __dump_to_json(self, content, file):
+        with open(file, 'w') as config_file:
+            json.dump(content, config_file)
+
+    def set_aws_profile(self, profile):
+        self.aws_profile = profile
+
+    def add_region(self, region):
+        if self.cloud == 'aws':
+            # If no profile was specified, select the default
+            if not hasattr(self, 'aws_profile'):
+                self.aws_profile = 'aws'
+
+            new_provider = {'region': region, 'alias': region, 'profile': self.aws_profile}
+
+        self.providers['provider'][self.cloud].append(new_provider)
+
+    def add_instance(self, region, image_id, instance_type=''):
+        aliases = [provider["alias"] for provider in self.providers["provider"][self.cloud]]
+        if not region in aliases:
+            print('Cannot add an instance if region provider is not set up')
+            exit(1)
+
+        if self.cloud == 'aws':
+            if not instance_type:
+                instance_type = 't2.micro'
+
+            instance_name = f'{region}-{instance_type}'.replace('.', '-')
+            new_instance = {
+                'instance_type': instance_type,
+                'ami': image_id,
+                'provider': f'aws.{region}',
+                'tags': {'name': instance_name},
+            }
+            self.resources['resource']['aws_instance'][instance_name] = new_instance
+            return instance_name
+
+    def get_instances(self):
+        if self.cloud == 'aws':
```

Comment:
Is this really a limitation for us to get instances that Terraform created? I guess we will discover it when we implement other cloud providers.

Anyways, the output of the `terraform show --json` command execution shouldn't be conditioned by the cloud provider IMO.

Reply:
Mhh, I think it will get dirty quickly if we don't have condition for clouds, on aws `instance_id` might be called differents. The same goes for other data points. Key names are probably specific to the cloud provider